core/tauri/src/updater/core.rs

```diff
@@ -973,26 +957,45 @@ fn copy_files_and_run<R: Read + Seek>(
       let powershell_install_res = powershell_cmd.spawn();
       if powershell_install_res.is_err() {
         // fallback to running msiexec directly - relaunch won't be available
-        // we use this here in case powershell fails in an older machine somehow
-        let msiexec_path = system_root.as_ref().map_or_else(
-          |_| "msiexec.exe".to_string(),
-          |p| format!("{p}\\System32\\msiexec.exe"),
-        );
-        let _ = Command::new(msiexec_path)
-          .arg("/i")
-          .arg(msi_path)
-          .args(installer_args)
-          .arg("/promptrestart")
-          .spawn();
+        installer_args_common.extend(config.tauri.updater.windows.install_mode.msiexec_args().iter().map(OsStr::new));
+        installer_args_common.push(OsStr::new("/promptrestart"));
+      } else {
+        exit(0);
       }
-
-      exit(0);
+    } else {
+      continue;
+    }
+    let file = encode_wide(found_path.as_os_str());
+    let parameters = encode_wide(installer_args_common.join(OsStr::new(" ")).as_os_str());
+
+    let ret = unsafe {
+      ShellExecuteW(
+        HWND(0),
+        w!("open"),
+        PCWSTR(file.as_ptr()),
+        PCWSTR(parameters.as_ptr()),
+        PCWSTR::null(),
+        std::mem::transmute(SW_SHOW),
```

Comment:
why use transmute here? `SW_SHOW.0` should be enough
```suggestion
        SW_SHOW.0,
```